tests/test_supply_curve_sc_aggregation.py

```diff
@@ -711,6 +712,203 @@ def test_format_res_fpath_with_year_pattern():
         assert _format_res_fpath(config) == {"res_fpath": tf.format(2010)}
 
 
+@pytest.mark.parametrize("zone_config,max_workers,pre_extract_inclusions", [
+    ("one_full", None, False),
+    (1, None, False),
+    (2, None, False),
+    (3, None, False),
+    (1, 1, False),  # test with run_serial
+    (1, None, True),  # test with pre_extract_exclusions
+])
+def test_agg_zones(zone_config, max_workers, pre_extract_inclusions):
+    """Test sc aggregation with zones within each sc site."""
+
+    resolution = 64
+    gids = [1, 2, 3]
+
+    with tempfile.TemporaryDirectory() as td:
+        excl_temp = os.path.join(td, "excl.h5")
+        shutil.copy(EXCL, excl_temp)
+        with SupplyCurveExtent(excl_temp, resolution=resolution) as sc:
+            slice_lookup = sc.get_slice_lookup(gids)
+
+        with h5py.File(excl_temp, 'a') as f:
+            shape = f[LATITUDE].shape
+            attrs = dict(f['ri_smod'].attrs)
+            profile = json.loads(attrs["profile"])
+            profile["dtype"] = "uint32"
+            profile["nodata"] = 0
+            attrs["profile"] = json.dumps(profile)
+            data = np.zeros(shape, dtype=np.uint32)
+            if zone_config == "one_full":
+                # each entire cell is one zone
+                for gid, gid_slice in slice_lookup.items():
+                    data[gid_slice] = gid + 10
```

Comment:
Would it be worth having a separate test with multiple zones per SC point? Just to make sure we get repeated `sc_point_gid` outputs with different zone ID's?  Or is that too cumbersome to set up?

Reply:
Oooops! I guess I missed the other branch of the if statement somehow. 
Thanks for being so thorough!